typescript/src/doc_functionality/general-utils.ts

```diff
@@ -52,4 +53,42 @@ export function generateCustomCSSHash(configuration: any): string {
         }, {});
 
     return hash(JSON.stringify(filteredConfiguration));
-}
\ No newline at end of file
+}
+
+
+interface VersionObject {
+  id: string;
+  name: string;
+  description: string;
+  version: string;
+  changeLog: string | null;
+  isReadOnly: boolean;
+  isSharedDraft: boolean;
+}
+
+interface VersionWithKey {
+  original: VersionObject;
+  key: string | null;
+}
+
+export const sortVersionsBySemver = (versions: VersionObject[]): VersionObject[] => {
+  const versionsWithKeys: VersionWithKey[] = versions.map((version) => ({
+    original: version,
+    key: version.version.match(/\d+\.\d+\.\d+(-[a-zA-Z0-9_]+)?/)?.[0] || null,
+  }));
+
+  const validVersions = versionsWithKeys.filter((item) => item.key);
+
+  const sortedValidVersions = semverSort
+    .desc(validVersions.map((item) => item.key!))
+    .map(
+      (sortedKey) =>
+        validVersions.find((item) => item.key === sortedKey)!.original
+    );
+
+  const nonSemverEntries = versionsWithKeys
+    .filter((item) => !item.key)
+    .map((item) => item.original);
+
```

Comment:
In the `cloud` we sort non semver versions by `createdAt`

Reply:
We sadly don't have that information in the exporter currently